### PR TITLE
heron package manager tool

### PR DESCRIPTION
**Pull request by nlu90**
This is a initial PR for a versioned package management tool for Heron. The works to be reviewed in this PR:
1. A set of interfaces for managing versioned packages
2. A simple local file system based example implementation

### CODE REVIEW

File: heron/package/src/python/common/parser.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    command_help = namespace['help-command']
+
+    if command_help == 'help':
+        parser.print_help()
+        return True
+
+    subparser = get_subparser(parser, command_help)
+    if subparser:
+        print(subparser.format_help())
+        return False
```
Comment (objmagic) [inline heron/package/src/python/common/parser.py]: please follow https://github.com/twitter/heron/pull/1086/files#diff-e6e2ba32ba0d8f08be1e918dc5d8f95aR64

**Comment by kramasamy** [comment]
can we call this packer itself instead of package? the external tool that will be shipped will be called ```heron-packer```

File: heron/package/src/python/BUILD [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package(default_visibility = ["//visibility:public"])
```
Comment (kramasamy) [inline heron/package/src/python/BUILD]: Can you add a lot of comments so that we can understand?

**Comment by kramasamy** [comment]
@nlu90 - can you please add several unit tests?

**Comment by nlu90** [comment]
@kramasamy I refactored the code and added many comments.

I'll add unit test in the next commit

**Comment by kramasamy** [comment]
Please add unit tests

**Comment by nlu90** [comment]
@kramasamy unit tests for LocalBlobstore and LocalMetastore are added

File: heron/packer/src/python/store/metastore.py [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+
+    meta_info = self._load_meta_file()
+
+    if role in meta_info and pkg_name in meta_info[role]:
+      version_map = meta_info[role][pkg_name]
+      for key, value in version_map.items():
+        # ignore tag entries in the version_map
+        if key != constants.LIVE and \
+           key != constants.LATEST and \
+           key != constants.COUNTER:
```
Comment (objmagic) [inline heron/packer/src/python/store/metastore.py]: maybe `key not in [constants.LIVE, constants.LATEST, constants.COUNTER]`?

**Comment by objmagic** [comment]
#1204 has just been merged. Please first merge with master and change to **use log module under [`heron/common/src/python/utils`](https://github.com/twitter/heron/blob/master/heron/common/src/python/utils/log.py)**. You can refer to [`heron-explorer`](https://github.com/twitter/heron/blob/master/heron/explorer/src/python/main.py#L31) to see how to config and use this log module.

**Comment by nlu90** [comment]
Will close this PR for it's opened too long ago and now we have the downloaders spi.

**Pull request closed without merging**